server/src/test/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapperTests.java

```diff
@@ -493,6 +492,67 @@ public void testDocumentsWithIncorrectDims() throws Exception {
         }
     }
 
+    /**
+     * Test that max dimensions limit for float dense_vector field
+     * is 2048 as defined by {@link XKnnFloatVectorField}
+     */
+    public void testMaxDimsFloatVector() throws IOException {
```

Comment:
This test verifies that we support 2048, but not necessarily that it's the max number of dimensions? Do we already have a test that verifies that we can't go higher than that?

Reply:
Yes, we have a test called `testDims()`, but before we only checked the limit non-indexed vectors. I've added the test for indexed vectors.